crates/ethrpc/src/current_block/mod.rs

```diff
@@ -134,6 +135,33 @@ pub async fn current_block_stream(url: Url, poll_interval: Duration) -> Result<C
     Ok(receiver)
 }
 
+/// Returns a reasonable poll interval for the given network.
+async fn poll_interval(web3: &Web3) -> Result<Duration> {
+    let chain = web3.eth().chain_id().await?;
+
+    let block_time = match chain.as_u64() {
+        1 => Duration::from_millis(12_000),
+        100 => Duration::from_millis(12_000),
```

Comment:
gnosis chain is 5s i think https://gnosisscan.io/blocks

Reply:
agree with mateo we should consider documenting the source for all these numbers

Reply:
True, forgot to push the 5s last night.
Will also add comments with links to the block frequency.